### ISSUE TITLE
Signature of superRefine() changed from zod@3.21 to zod@3.22

### ISSUE THREAD

**Issue by paulsmithkc** [colinhacks/zod#2665]
With `zod@3.21` the following code was valid:

```
import { z } from "zod";

const schema = z.object({
  name: z.string(),
  value: z.string(),
});

const schemaRefine = schema.superRefine((val, _ctx) => {
  return val.value !== 'INVALID';
});
```

After upgrading to `zod@3.22` we are seeing the following typescript error:

```
Type 'boolean' is not assignable to type 'void | Promise<void>'.
```

**Comment by paulsmithkc** [comment]
In `zod@3.21` the relevant typedefs were as follows:

```
superRefine<RefinedOutput extends Output>(refinement: (arg: Output, ctx: RefinementCtx) => arg is RefinedOutput): ZodEffects<this, RefinedOutput, Input>;
superRefine(refinement: (arg: Output, ctx: RefinementCtx) => void): ZodEffects<this, Output, Input>;
```

In `zode@3.22` the relevant typedefs are as follows:

```
superRefine<RefinedOutput extends Output>(refinement: (arg: Output, ctx: RefinementCtx) => arg is RefinedOutput): ZodEffects<this, RefinedOutput, Input>;
    superRefine(refinement: (arg: Output, ctx: RefinementCtx) => void | Promise<void>): ZodEffects<this, Output, Input>;
```

**Comment by colinhacks** [comment]
Thanks, should be fixed in 3.22.2.
